internal/util/diff/diff.go

```diff
@@ -301,3 +320,17 @@ func (pg defaultPkgGetter) GetPkg(repo, path, ref string) (string, error) {
 	err = cmdGet.Run()
 	return dir, err
 }
+
+// shortSha returns a shortened version of a commit SHA
+func shortSha(sha string) string {
+	return sha[0:int(math.Min(float64(len(sha)), 7))]
+}
+
+// stageDirectory creates a subdirectory of the provided path for temporary operations
+// path is the parent staged directory and should already exist
+// subpath is the subdirectory that should be created inside path
+func stageDirectory(path, subpath string) (string, error) {
```

Comment:
seems like there is a handy function but earlier there is an ioutils.TempDir call still.  Is this supposed to be the one place that standardizes our calls?  Why not use it?

Reply:
I create a single parent temp directory (the `ioutils.TempDir` call you are referring to) and then inside that generated directory create additional staged directories. This function was written so the returned types match the signature of the `TempDir` function so that they may be swapped interchangeably as needed. The `TempDir` call still serves a purpose and does additional work to ensure conflicts are avoided when creating the directory. Once we have an empty working directory we can interact with it with less safeguards so random strings and additional safety checks were removed from this wrapper.

This change means that `/tmp/kpt-foo-12348765` will now appear as `/tmp/kpt-12348765/foo` and `/tmp/kpt-12348765/bar` for example.

This change also means that duplicate logic to cleanup our staging directories was removed and located in a single spot that cleans up the top level staging directory.

Reply:
To expand a little: if we remove the creation of a temp directory entirely then if the directory cleanup fails OR kpt exits prior to cleaning up the directories then diffs may become broken until that is fixed by the user/OS. Maintaining this behavior helps avoid that situation.